### ISSUE TITLE
Wolfenstein 2 assert in register allocation

### ISSUE THREAD

**Issue by tarceri** [daniel-schuermann/mesa#25]
Asserts on:

`assert(ctx.assignments[affinities[definition.tempId()]].second == phi->getDefinition(0).regClass());`

I'm not entirely sure what the allocation code is doing but it hits the assert when processing block 131 so I've included a dump of hopefully relevant code below:

```
BB129
/* logical preds: BB128, / linear preds: BB128,  */
	p_logical_start
	 s1: %1967,   b: %1968:scc = s_add_i32 0x20, %2
	 s2: %1969 = p_create_vector %1967, 0xffff8000
	 s4: %1971 = s_load_dwordx4 %1969, 0x0
	 v1: %1966 = v_lshrrev_b32 0x10, %107
	 v1: %1970 = v_lshlrev_b32 0x6, %1966
	 v3: %1972 = buffer_load_dwordx3 %1970, %1971, 0x0
	 v1: %1973,  v1: %1974,  v1: %1975 = p_split_vector %1972
	 v1: %1976 = v_add_f32 %115, %1973
	 v1: %1977 = v_add_f32 %117, %1974
	 v1: %1978 = v_add_f32 %119, %1975
	p_logical_end
	p_branch BB131
BB130
/* logical preds: BB128, / linear preds: BB128,  */
	p_logical_start
	p_logical_end
	p_branch BB131
BB131
/* logical preds: BB129, BB130, / linear preds: BB129, BB130,  */
	 v1: %1981 = p_phi %1978, %119
	 v1: %1980 = p_phi %1977, %117
	 v1: %1979 = p_phi %1976, %115
	p_logical_start
	p_logical_end
	p_cbranch_z %1865:scc BB133, BB132
BB132
/* logical preds: BB131, / linear preds: BB131,  */
	p_logical_start
	 s1: %1983,   b: %1984:scc = s_add_i32 0x20, %2
	 s2: %1985 = p_create_vector %1983, 0xffff8000
	 s4: %1988 = s_load_dwordx4 %1985, 0x0
	 v1: %1982 = v_lshrrev_b32 0x10, %107
	 v1: %1986 = v_lshlrev_b32 0x6, %1982
	 v1: %1987 = v_add_u32 0x10, %1986
	 v3: %1989 = buffer_load_dwordx3 %1987, %1988, 0x0
	 v1: %1990,  v1: %1991,  v1: %1992 = p_split_vector %1989
	 v1: %1993 = v_add_f32 %109, %1990
	 v1: %1994 = v_add_f32 %111, %1991
	 v1: %1995 = v_add_f32 %113, %1992
	p_logical_end
	p_branch BB134
BB133
/* logical preds: BB131, / linear preds: BB131,  */
	p_logical_start
	p_logical_end
	p_branch BB134
BB134
/* logical preds: BB132, BB133, / linear preds: BB132, BB133,  */
	 v1: %1998 = p_phi %1995, %113
	 v1: %1997 = p_phi %1994, %111
	 v1: %1996 = p_phi %1993, %109
	p_logical_start
	p_logical_end
	p_cbranch_z %1899:scc BB145, BB135
BB135
/* logical preds: BB134, / linear preds: BB134,  */
	p_logical_start
	 s1: %1999,   b: %2000:scc = s_add_i32 0x60, %2
	 s2: %2001 = p_create_vector %1999, 0xffff8000
	 s4: %2006 = s_load_dwordx4 %2001, 0x0
	 s1: %2002,  s1: %2003:scc = s_lshl_b32 %1857, 0x2
	 v1: %2005 = p_parallelcopy 0x1
	 v1: %2007 = buffer_atomic_add undef, %2006, %2002, %2005
	p_logical_end
	 s2: %2008 = s_mov_b64 %0:exec
	p_branch BB136
BB136
/* logical preds: BB135, BB143, / linear preds: BB135, BB143,  */
	 s2: %2408 = p_linear_phi %2411, %2430
	 v1: %2010 = p_phi %2007, %2011
	 s2: %2009 = s_mov_b64 %0:exec
	p_logical_start
	 v1: %2012 = v_add_u32 0x1, %2007
	 v1: %2013 = v_min_i32 %1860, %2012
	 s2: %2014 = v_cmp_ge_i32 %2010, %2013
	p_logical_end
	 s2: %2015,   b: %2016:scc,  s2: %2017:exec = s_and_saveexec_b64 %2014
	 s2: %2018,   b: %2019:scc = s_andn2_b64 %2015, %2014
	p_cbranch_z %2017:exec BB139, BB137
BB137
/* logical preds: BB136, / linear preds: BB136,  */
	p_logical_start
	p_logical_end
	 s2: %2020,   b: %2021:scc = s_andn2_b64 %2009, %0:exec
	 s2: %0:exec = s_mov_b64 0x0
	p_cbranch_z %2021:scc BB144, BB138
BB138
/* logical preds: / linear preds: BB137,  */
	p_branch BB140
BB139
/* logical preds: / linear preds: BB136,  */
	p_branch BB140
BB140
/* logical preds: / linear preds: BB138, BB139,  */
	 s2: %2023 = s_mov_b64 %0:exec
	p_cbranch_z %2018:exec BB142, BB141
BB141
/* logical preds: BB136, / linear preds: BB140,  */
	p_logical_start
	p_logical_end
	p_branch BB143
BB142
/* logical preds: / linear preds: BB140,  */
	p_branch BB143
BB143
/* logical preds: BB141, / linear preds: BB141, BB142,  */
	 s2: %2430 = p_linear_phi %2408, %2408
	 s2: %0:exec,   b: %2024:scc = s_or_b64 %0:exec, %2023
	p_logical_start
	 s1: %2027,   b: %2028:scc = s_add_i32 0x70, %2
	 s2: %2029 = p_create_vector %2027, 0xffff8000
	 s4: %2031 = s_load_dwordx4 %2029, 0x0
	 v1: %2025 = v_add_u32 %1856, %2010
	 v3: %2026 = p_create_vector %1979, %1980, %1981
	 v1: %2030 = v_lshlrev_b32 0x5, %2025
	buffer_store_dwordx3 %2030, %2031, 0x0, %2026
	 v1: %2032 = v_add_u32 0x10, %2030
	 v4: %2033 = p_create_vector %1996, %1997, %1998, %107
	buffer_store_dwordx4 %2032, %2031, 0x0, %2033
	 v1: %2011 = v_add_u32 0x1, %2010
	p_logical_end
	p_branch BB136
```

**Comment by pendingchaos** [comment]
It's a regression from cf33038c9074a7335c9d62ee44b2047188ca5686. I believe the RA is creating a phi with a mixture of `scc` and `s1` as the register class.

**Comment by pendingchaos** [comment]
Should be fixed in 15cd277d130aaeca01d26710c353b4df148674fc.
